### DISCUSSION TITLE
How do I connect to a zeptoforth Raspberry Pi Pico from Windows?

### DISCUSSION THREAD

**Discussion by znmeb** [tabemann/zeptoforth#29] [Q&A] [answered]
I have a Pico connected to Windows via a USB cable. I loaded the `zeptoforth_full-0.60.1.uf2` onto it and started up [https://tabemann.github.io/zeptocomjs/zeptocom.html](https://tabemann.github.io/zeptocomjs/zeptocom.html) in Chrome. But Windows isn't seeing a serial device in the Device Manager and the zeptocom application isn't seeing a port to connect to. Is there another step I have to do to make this work?

**Comment by znmeb** [comment]
I found it - the step I was missing was connecting to the Pico via the UART. I have another Pico running `picoprobe` that's connected to the one running `zeptoforth` and that works. Is there no USB serial mode in `zeptoforth`?

**Reply by tabemann** [reply]
zeptoforth only currently has USB-serial on boards that have separate USB-serial chips on them, i.e. the STM32F746 and STM32L476 DISCOVERY. Adding USB-serial support to zeptoforth on other platforms is a future project, after I either complete or give up on my Wio RP2040 ESP8285 radio project.

**Reply by znmeb** [reply]
That's OK - the new debug probe is only $12US and it does both SWD and UART over USB. It's a packaged version of what I'm currently doing with a second Pico.

**Reply by tabemann** [reply, marked as answer]
Just so you know, USB CDC console support has been added for the RP2040 in `full_usb` builds starting with release 1.0.0.
